sponsor-dapp-v2/src/components/ManagePositionButton.js

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+
+import classNames from "classnames";
+
+import { useTokenPreapproval, useSettle, useExpire, useIsContractSponsor } from "lib/custom-hooks";
+
+function ManagePositionButton(props) {
+  const { contractAddress, identifier, history } = props;
+
+  const { ready: approveReady, approveTokensHandler, isApproved, isLoadingApproval } = useTokenPreapproval(
+    "TestnetERC20",
+    contractAddress
+  );
+
+  const { ready: settleReady, canCallSettle, settleHandler, isLoadingSettle } = useSettle(contractAddress);
+  const { ready: expireReady, canCallExpire, expireHandler, isLoadingExpire } = useExpire(contractAddress, identifier);
+  const isContractSponsor = useIsContractSponsor(contractAddress);
+  const ready = approveReady && settleReady && expireReady && isContractSponsor !== undefined;
+
+  if (!ready) {
+    return (
+      <a
+        href="test"
+        className={classNames("btn", {
+          disabled: true
+        })}
+      >
+        Loading
+      </a>
+    );
+  }
+
+  let buttonDetails;
+
+  if (canCallSettle) {
+    buttonDetails = {
+      disabled: false,
+      isLoading: isLoadingSettle,
+      text: "Settle Contract",
+      handler: settleHandler
+    };
+  } else if (canCallExpire) {
+    buttonDetails = {
+      disabled: false,
+      isLoading: isLoadingExpire,
+      text: "Expire Contract",
+      handler: expireHandler
+    };
+  } else if (isContractSponsor && !isApproved) {
```

Comment:
@ptare one minor note here: Not sure if we should prefer showing the approval button or the expiry/settlement buttons if both actions are possible.

Reply:
Preferring settle and expire seem right to me. Even if we allow them to approve first, the dapp won't let them deposit or create tokens anyway because they can't get to the Manage Positions page.